jobs/config.json

```diff
@@ -5275,6 +5275,24 @@
       "sig-cluster-lifecycle"
     ]
   },
+  "ci-kubernetes-e2e-kubeadm-gce-1-6-on-1-7": {
+    "args": [
+      "--cluster=",
+      "--env-file=jobs/platform/gce.env",
+      "--env-file=jobs/ci-kubernetes-e2e-kubeadm-gce-1-6-on-1-7.env",
+      "--kubeadm=ci",
+      "--kubernetes-anywhere-kubernetes-version=latest-1.6",
+      "--deployment=kubernetes-anywhere",
+      "--mode=local",
+      "--timeout=300m",
+      "--extract=ci/latest-1.6",
```

Comment:
So, the `--extract` flag ends up determining where to get compiled Kubernetes binaries from. It gets passed around and ultimately interpreted and used by [`kubetest`](https://github.com/kubernetes/test-infra/blob/616f1aa9c9e4e90db05fd9a275450a2ca4abfcf1/kubetest/main.go#L88). In our case, the most important is the compiled e2e tests to run. Since they can be added/changed/removed between releases, it's important that we're using 1.6-based Conformance tests, so `--extract=ci/latest-1.6` means to look at the latest CI build for 1.6 and fetch its binary artifacts. You can see the extraction code in [`extract.go`](https://github.com/kubernetes/test-infra/blob/616f1aa9c9e4e90db05fd9a275450a2ca4abfcf1/kubetest/extract.go).

The fact that we're actually using kubeadm 1.7 comes from the prow configuration, which is what triggers the test. It's not obvious in the diff, because this new job gets chained off as the child of an existing job, but if you crawl up the `config.yaml` diff,  you can see that the parent job is `ci-kubernetes-bazel-test-1-7`, whose parent is `ci-kubernetes-bazel-build-1-7`. The `ci-kubernetes-bazel-build-1-7` job is configured to only run on the release-1.7 branch:

    - name: ci-kubernetes-bazel-build-1-7
      branches:
      - release-1.7

Then, at the end of its job definition, it has:

     run_after_success:
     - name: ci-kubernetes-bazel-test-1-7

So once the bazel build finishes, it triggers the bazel tests. Once those finish, they trigger two jobs (since `run_after_success` is a list):

     run_after_success:
     - name: ci-kubernetes-e2e-kubeadm-gce-1-7
       <snip>
     - name: ci-kubernetes-e2e-kubeadm-gce-1-6-on-1-7

These new entries don't need to repeat that they only run on the `release-1.7` branch because the only way they get triggered in the first place is if the parent jobs succeed, and those only trigger on `release-1.7` commits.

These jobs use the `.deb` files built and uploaded by the parent bazel build job, so they'll be using 1.7 versions of kubeadm/kubelet/kubectl on the GCE VMs. The `--kubernetes-anywhere-kubernetes-version=latest-1.6` flag in `config.json` ends up being fed into `kubeadm init --kubernetes-version=latest-1.6` to use the latest 1.6 images of the control plane.

Does that clear things up?